sched/irq/irq_csection.c

```diff
@@ -355,7 +355,8 @@ irqstate_t enter_critical_section(void)
                    * request.
                    */
 
-                  up_irq_restore(ret);
+                  up_irq_enable();
+                  up_irq_save();
```

Comment:
@masayuki2009 could you please take a look?

Reply:
I'll try this PR later.